app/src/main/java/org/dhis2/Bindings/RuleExtensions.kt

```diff
@@ -293,6 +293,8 @@ fun List<TrackedEntityDataValue>.toRuleDataValue(
                     optionRepository.byOptionSetUid().eq(de.optionSetUid()).byCode().eq(value).one()
                         .blockingGet().name()
             }
+        } else if (de.valueType()!!.isNumeric) {
```

Comment:
can value type be null if it does not match any value?

Reply:
Nop, all dataElements and trackedEntityAttributes have a valueType. Web does not let you create one without this variable.